packages/chat-stateful-client/src/ResourceDownloadQueue.ts

```diff
@@ -95,14 +90,24 @@ export class ResourceDownloadQueue {
       }
       for (const attachment of attachments) {
         if (attachment.previewUrl && attachment.attachmentType === 'image') {
-          const blobUrl = await operation(attachment.previewUrl, this._credential);
+          const blobUrl = await this.downloadResource(operation, attachment.previewUrl);
           message.resourceCache[attachment.previewUrl] = blobUrl;
         }
       }
     }
 
     return message;
   }
+
+  private async downloadResource(operation: ImageRequest, url: string): Promise<string> {
+    let blobUrl = URL.createObjectURL(new Blob());
+    try {
+      blobUrl = await operation(url, this._credential);
+    } catch (error) {
+      console.log('Downloading Resource error: ', error);
```

Comment:
This error is no longer surfaced up the call stack. Is this behaviour change intended?
If it fails, a blobUrl is passed back to the caller with them having no knowledge of the error state.

Reply:
the error was never surfaced to the call stack, line 107 is exactly what we currently have, i just moved it down to child function